src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -139,19 +141,25 @@ def histogram_finish(job_id, retry):
     """
     This is the bottom-half of the function.
     """
-    def dict_to_array(d):
-        result = []
-        for k, v in d.iteritems():
-            [k1, k2] = map(int, re.sub('[^0-9,]', '', k).split(','))
-            result.append(((k1, k2), v))
-        return result
-
     host = settings.GEOP['host']
     port = settings.GEOP['port']
 
-    data = sjs_retrieve(host, port, job_id, retry)
+    sjs_result = sjs_retrieve(host, port, job_id, retry)
+    data = {make_tuple(key[4:]): val for key, val in sjs_result.items()}
```

Comment:
Unclear what format conversion is going here, could use a comment to clear up why it's necessary.